### PR TITLE
[aot] Parallelize compilation

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

**Comment by ludovic-henry** [comment]
cc @jonpryor

File: src/Xamarin.Android.Build.Tasks/Tasks/Aot.cs [hunk 1]
```diff
@@ -224,6 +225,44 @@ static int GetNdkApiLevel(string androidNdkPath, string androidApiLevel, Android
 
 			var nativeLibs = new List<string> ();
 
+			CancellationTokenSource cts = null;
```
Comment (jonpryor) [inline src/Xamarin.Android.Build.Tasks/Tasks/Aot.cs]: Could you please place this into a separate method? Help keep method size "reasonable"?

**Comment by jonpryor** [comment]
@ludovic-henry: This [fails to build](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/315/consoleFull#-204209783650d48019-3ad7-484e-a279-755473c6305f):

    Tasks/Aot.cs(245,143): error CS1525: Unexpected symbol `='

This line causes the CS1525:

    System.Threading.Tasks.Parallel.ForEach (GetAotConfigs (), new System.Threading.Tasks.ParallelOptions { System.Threading.CancellationToken = cts.Token },

The problem is the `new ParallelOptions { System.Threading.CancellationToken = ...`. It should instead be:

    System.Threading.Tasks.Parallel.ForEach (GetAotConfigs (), new System.Threading.Tasks.ParallelOptions { CancellationToken = cts.Token },

Additionally, I like to keep line widths "short", which makes things easier when doing e.g. side-by-side diffs, or when changing any particular line for a bug fix (line lines "hide" changes). As such:

1. Please add a `using System.Threading;` and `using System.Threading.Tasks;` instead of fully qualifying all the type names. Namespaces should be sorted, thus:

        using System;
        using System.Collections.Generic;
        using System.Collections.Specialized;
        using System.Diagnostics;
        using System.IO;
        using System.Linq;
        using System.Reflection;
        using System.Threading;
        using System.Threading.Tasks;

        using Microsoft.Build.Framework;
        using Microsoft.Build.Utilities;

2. Introduce intermediate variables to assist in shorter line widths.

For example:

    cts         = new CancellationTokenSource ();
    var options = new ParallelOptions {
        CancellationToken = cts.Token,
    };
    Parallel.ForEach (GetAotConfigs (), options, config => {
            ...
    });

This has the additional nice affect that if we need additional parallel options, they are easily added with minimal line changes and "noise."

**Comment by ludovic-henry** [comment]
@jonpryor I fixed the compilation failure and updated according to your comment

**Pull request merged**